include/enrico/coupled_driver.h

```diff
@@ -42,7 +42,10 @@ class CoupledDriver {
   virtual void set_heat_source() {}
 
   //! Update the temperature for the neutronics solver
-  virtual void update_temperature() {}
+  virtual void update_temperature() final;
+
+  //! Set the temperature in the neutronics solver
+  virtual void set_temperature() {};
```

Comment:
Maybe this should this be pure virtual?

Reply:
It could be pure virtual, the reason why I didn't make `set_temperature` and `set_heat_source` pure virtual was that some surrogate drivers might not solve for "all" the physics. For example, the `SurrogateHeatDriver` doesn't solve for density at all, so if we had a pure virtual `set_density` method, we'd need to define the method as empty. 

Basically, I thought a decent default would be to do nothing, but I'm also ok with making it pure virtual - let me know what you prefer!

Reply:
Good point! let's leave it as is